src/h5web/visualizations/heatmap/ColorMapSelector.tsx

```diff
@@ -1,137 +1,83 @@
-import React, { CSSProperties } from 'react';
-import Select, {
-  components,
-  OptionProps,
-  IndicatorProps,
-  SingleValueProps,
-} from 'react-select';
-import { FiChevronDown } from 'react-icons/fi';
-import shallow from 'zustand/shallow';
-import { generateCSSLinearGradient, convertToOptions } from './utils';
-import {
-  INTERPOLATORS,
-  MOST_USED,
-  SINGLE_HUE,
-  MULTI_HUE,
-  CYCLICAL,
-  DIVERGING,
-} from './interpolators';
-import { useHeatmapConfig } from './config';
-import styles from './HeatmapToolbar.module.css';
-import { ColorMapOption } from './models';
-import { customThemeForSelect } from '../shared/utils';
+import React, { ReactElement } from 'react';
+import { Wrapper, Button, Menu, MenuItem } from 'react-aria-menubutton';
 
-const colorMapOptions = [
-  {
-    label: 'Common',
-    options: convertToOptions(MOST_USED),
-  },
-  {
-    label: 'Single hue',
-    options: convertToOptions(SINGLE_HUE),
-  },
-  {
-    label: 'Multi hue',
-    options: convertToOptions(MULTI_HUE),
-  },
-  {
-    label: 'Cyclical',
-    options: convertToOptions(CYCLICAL),
-  },
-  {
-    label: 'Diverging',
-    options: convertToOptions(DIVERGING),
-  },
-];
+import { MdArrowDropDown } from 'react-icons/md';
+import { useWindowSize } from 'react-use';
+import styles from './ColorMapSelector.module.css';
+import { INTERPOLATOR_GROUPS, INTERPOLATORS } from './interpolators';
+import { ColorMap, D3Interpolator } from './models';
+import { generateCSSLinearGradient } from './utils';
 
-function ColorMapLabel(props: ColorMapOption): JSX.Element {
-  const { label, value } = props;
+const MENU_IDEAL_HEIGHT = 320; // 20rem
+const MENU_TOP = 87; // HACK: height of breadcrumbs bar + height of toolbar
+const MENU_BOTTOM = 16; // offset from bottom of viewport
```

Comment:
These constants are for the dynamic height feature. We need to know the top position of the menu on the viewport, but `useMeasure` returns parent-relative positions... I decided to hard code `MENU_TOP` for now, hoping that `react-use` will improve its hook in the future.